game-core/src/main/java/games/strategy/triplea/delegate/FinishedBattle.java

```diff
@@ -119,8 +119,8 @@ public void removeAttack(final Route route, final Collection<Unit> units) {
         m_isAmphibious = !m_amphibiousAttackFrom.isEmpty();
       }
     }
-    for (final Unit dependence : m_dependentUnits.keySet()) {
-      final Collection<Unit> dependent = m_dependentUnits.get(dependence);
+    for (final Map.Entry<Unit, Collection<Unit>> unitCollectionEntry : m_dependentUnits.entrySet()) {
+      final Collection<Unit> dependent = unitCollectionEntry.getValue();
       dependent.removeAll(units);
     }
```

Comment:
It would make sense to iterate over `values()` here
Also there's no need to create a new variable